gateway/ln-gateway/src/lib.rs

```diff
@@ -1123,6 +1123,55 @@ impl Gateway {
         Ok(())
     }
 
+    pub async fn handle_connect_to_peer_msg(
+        &self,
+        ConnectToPeerPayload { pubkey, host }: ConnectToPeerPayload,
+    ) -> Result<()> {
+        let context = self.get_lightning_context().await?;
+        context.lnrpc.connect_to_peer(pubkey, host).await?;
+        Ok(())
+    }
+
+    pub async fn handle_get_funding_address_msg(&self) -> Result<Address> {
+        let context = self.get_lightning_context().await?;
+        let response = context.lnrpc.get_funding_address().await?;
+        Address::from_str(&response.address)
+            .map(|address| address.assume_checked())
+            .map_err(|e| GatewayError::ResponseParseError(e.into()))
+    }
+
+    pub async fn handle_open_channel_msg(
+        &self,
+        OpenChannelPayload {
+            pubkey,
+            channel_size_sats,
+            push_amount_sats,
+        }: OpenChannelPayload,
+    ) -> Result<()> {
+        let context = self.get_lightning_context().await?;
+        context
+            .lnrpc
+            .open_channel(pubkey, channel_size_sats, push_amount_sats)
+            .await?;
+        Ok(())
+    }
+
+    pub async fn handle_wait_for_chain_sync_msg(
+        &self,
+        WaitForChainSyncPayload {
+            block_height,
+            max_retries,
+            retry_delay,
+        }: WaitForChainSyncPayload,
+    ) -> Result<()> {
+        let context = self.get_lightning_context().await?;
+        context
+            .lnrpc
+            .wait_for_chain_sync(block_height, max_retries, retry_delay)
```

Comment:
Polling for block height can be implemented here instead of polling in each impl

Reply:
Makes sense. Before I do that, what are your thoughts about removing `wait_for_chain_sync()`, adding a `synced_block_height` field to `GatewayInfo`, then having the `wait-for-chain-sync` CLI command simply poll the gateway CLI's `info` command until the value of the new `synced_block_height` field is >= the target block height?